Use archive entry length instead of original length for EPUB positions

mickael-menu opened the pull request:
### Changed

* The default EPUB positions service now uses the archive entry length when available. [This is similar to how Adobe RMSDK generates page numbers](https://github.com/readium/architecture/issues/123).


A few downsides with this algorithm:
* if you explode a ZIP archive or serve it from the web, you won't get the same number of positions
* the encryption algorithm used impacts the number of positions (e.g. an encrypted book won't have exactly the same number of positions, same for ACS vs LCP books)

An alternative could be to keep the previous algorithm but change the page length value (which is 1024 at the moment).

(@aferditamuriqi Let's wait for the next call before merging this one)

aferditamuriqi reviewed (dismissed):
looks good and works as expected

mickael-menu commented:
I added an `EPUBPositionsService.ReflowableStrategy` enum to set which strategy to use. We just need to decide which one should be the default during next call. 

```swift
/// Strategy used to calculate the number of positions in a reflowable resource.
///
/// Note that a fixed-layout resource always has a single position.
public enum ReflowableStrategy {
    /// Use the original length of each resource (before compression and encryption) and split it by the given
    /// `pageLength`.
    case originalLength(pageLength: Int)

    /// Use the archive entry length (whether it is compressed or stored) and split it by the given `pageLength`.
    case archiveEntryLength(pageLength: Int)

    /// Strategy used by Adobe RMSDK: use the archive entry length and split it by 1024 bytes.
    /// See https://github.com/readium/architecture/issues/123
    public static var adobeRMSDK = archiveEntryLength(pageLength: 1024)
}
```

mickael-menu commented:
We agreed on last dev call to use this historical/RMSDK approach as the "recommended" and default one, until W3C comes up with a standard for positions calculation. We will revisit this issue then.
